src/test/tree/EigenPodManager.tree

```diff
@@ -0,0 +1,95 @@
+├── EigenPodManager Tree (*** denotes that integrationt tests are needed to validate path)
+├── when contract is deployed and initialized
+│   └── it should properly set storage
+├── when initialize called again
+│   └── it should revert
+├── when createPod called
+│   ├── given the user has already created a pod
+│   │   └── it should revert
+│   ├── given that the max number of pods has been deployed
+│   │   └── it should revert
+│   └── given the user has not created a pod
+│       └── it should deploy a pod
+├── when stake is called
+│   ├── given the user has not created a pod
+│   │   └── it should deploy a pod
+│   └── given the user has already created a pod
+│       └── it should call stake on the eigenPod
+├── when setMaxPods is called
+│   ├── given the user is not the pauser
+│   │   └── it should revert
+│   └── given the user is the pauser
+│       └── it should set the max pods
+├── when updateBeaconChainOracle is called
+│   ├── given the user is not the owner
+│   │   └── it should revert
+│   └── given the user is the owner
+│       └── it should set the beacon chain oracle
+├── when addShares is called
+│   ├── given that the caller is not the delegationManager
+│   │   └── it should revert
+│   ├── given that the podOwne address is 0
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that too many shares are withdrawn
+│   │   └── it should revert
+│   └── given that all of the above conditions are satisfied
+│       └── it should update the podOwnerShares
+├── when removeShares is called
+│   ├── given that the caller is not the delegationManager
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that too many shares are removed
+│   │   └── it should revert
+│   └── given that all of the above conditions are satisfied
+│       └── it should update the podOwnerShares
+├── when shares are withdrawn as tokens
+│   ├── given that the podOwner is address 0
+│   │   └── it should revert
+│   ├── given that the destination is address 0
+│   │   └── it should revert
+│   ├── given that the shares amount is negative
+│   │   └── it should revert
+│   ├── given that the shares is not a whole gwei amount
+│   │   └── it should revert
+│   ├── given that the current podOwner shares are negative
+│   │   ├── given that the shares to withdraw are greater than the negative shares of the podOwner
+│   │   │   └── it should set the podOwnerShares to 0 and decrement shares to withdraw by the share defecit
```

Comment:
I find this wording hard to parse. maybe
`given that the shares to withdraw are larger in magnitude than the shares of the podOwner` instead?
should we also explain that any remainder moves on to the 'withdraw restaked ETH from the eigenPod' step?

also minor typo: `defecit` => `deficit`